src/text/span.rs

```diff
@@ -332,6 +332,37 @@ impl<'a> Span<'a> {
     pub fn to_right_aligned_line(self) -> Line<'a> {
         self.into_right_aligned_line()
     }
+
+    /// Splits the span into two at the given width.
+    ///
+    /// The first span will contain the characters that fit within the given width, and the second
+    /// span will contain the remaining characters.
+    pub(crate) fn split_at_width(&'a self, width: usize) -> (Self, Self) {
```

Comment:
I think this might be a good overgeneralization example. We only need the right side here, but a bunch of stuff is done for the left side too.
This will be called every time a Line or Text is rendered and should not do stuff that's not important here. Maybe its optimized away by the compiler again, but why overcomplicate the code which is not needed in the first place?

Reply:
Mainly because there's a bug that's affecting users. This is a known good state that passes (most) tests from which we can definitely optimize / rewrite. Happy for this to be ripped out entirely if there's a simpler more obvious way to approach this. My constraints were working code, and it looks like it does the right thing.

Reply:
https://github.com/ratatui-org/ratatui/pull/1089/commits/d64f785e8705f7da37f9d392746a914086ea7093